src/main/kotlin/me/serce/solidity/lang/resolve/ref/SolImportPathReference.kt

```diff
@@ -97,6 +101,34 @@ class SolImportPathReference(element: SolImportPathElement) : SolReferenceBase<S
       }
     }
 
+    private fun remappingsFromFoundryConfigFile(file: VirtualFile): List<Pair<String, String>> {
+      val foundryConfigFile = file.findFileByRelativePath("foundry.toml")
+      val remappingsResult = arrayListOf<Pair<String, String>>()
+      if (foundryConfigFile != null) {
+        val mapper = TomlMapper()
+        val data = mapper.readValue(foundryConfigFile.readText(), Map::class.java)
+        data["profile"].let { profile ->
+          if (profile != null) {
+            (profile as LinkedHashMap<*, *>)["default"].let { profileDefault ->
+              if (profileDefault != null) {
+                (profileDefault as LinkedHashMap<*, *>)["remappings"].let { remappings ->
+                  if (remappings != null) {
+                    (remappings as ArrayList<*>).forEach { mapping ->
+                      val splitMapping = mapping.toString().trim('"').split("=")
+                      if (splitMapping.size == 2) {
+                        remappingsResult.add(Pair(splitMapping[0].trim(), splitMapping[1].trim()))
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      return remappingsResult
+    }
```

Comment:
```suggestion
    private fun remappingsFromFoundryConfigFile(file: VirtualFile): List<Pair<String, String>> {
      val foundryConfigFile = file.findFileByRelativePath("foundry.toml")
      if (foundryConfigFile == null) {
        return emptyList()
      }
      val mapper = TomlMapper()
      val data = mapper.readTree(foundryConfigFile.readText())
      val remappings = data.get("profile")?.get("default")?.get("remappings")
      if (remappings == null) {
        return emptyList()
      }
      return remappings //
        .filterIsInstance<TextNode>() //
        .map { //
          // each is e.g. "forge-std/=lib/forge-std/src/"
          it.textValue().trim('"').split("=") //
        } //
        .filter { it.size == 2 } //
        // "forge-std/" to "lib/forge-std/src/"
        .map { it[0].trim() to it[1].trim() }
    }
```

I think you can improve readability of this function by switching to JsonNode API and reducing nestedness

Reply:
I also added a try-catch above to make sure that we don't throw on malformed ones.

Reply:
We should consider caching them assuming it's not the case already

Reply:
I can see how to cache the values but I also need to see why there is no actualization when adding/updating remappings.txt 
For example when loading a project with foundry, sometimes there is no remappings.txt and the user need to generate it. So the IDE is doing his indexation then the user add a remappings but the IDE doesn't reload himself. The easier way is to restart the IDE.